### DISCUSSION TITLE
(v2 API) `selectAtom`: Type 'Read<T, never>' is not assignable to type 'Read<T>'

### DISCUSSION THREAD

**Discussion by vf-telwing** [pmndrs/jotai#1669] [Q&A] [answered]
I'm using the stable reference pattern for `selectAtom` (as seen [here](https://jotai.org/docs/utils/select-atom#hold-stable-references)) and I'm unsure of how to fix this typing error.

Here is a minimal reproduction: https://codesandbox.io/s/heuristic-dubinsky-f8vob1?file=/src/App.tsx

Any help would be much appreciated!

**Comment by dai-shi** [comment, marked as answer]
Sorry for the confusion.
When you use Jotai v2 API, you need to use it for everything.
This should fix it:
```diff
- import { selectAtom } from "jotai/utils";
+ import { selectAtom } from "jotai/vanilla/utils";
```

Once jotai v2.0.0 is released, this problem won't happen.

**Reply by vf-telwing** [reply]
No worries, that makes sense. Thanks for the quick reply!
